src/runtime/slot-composer.ts

```diff
@@ -182,10 +161,13 @@ export class SlotComposer {
       // which has been extracted from DOM.
       // TODO: FIXIT!
       if (eventlet.data && eventlet.data.key) {
-        const hostedConsumers = this.consumers.filter(c => c instanceof HostedSlotConsumer && c.transformationSlotConsumer === slotConsumer);
-        for (const hostedConsumer of hostedConsumers) {
-          if (hostedConsumer instanceof HostedSlotConsumer && hostedConsumer.storeId) {
-            const store = hostedConsumer.arc.findStoreById(hostedConsumer.storeId);
+        // We fire off multiple async operations and don't wait.
+        slotConsumer.hostedSlotContexts.forEach(async ctx => {
```

Comment:
nit - i think we agreed to use the conventional for loop instead of forEach

Reply:
True. I had to restructure a bit to maintain the fire and forget behavior that I think makes sense here, but done.